Lib/argparse.py

```diff
@@ -1360,7 +1360,7 @@ def __init__(self,
         self._defaults = {}
 
         # determines whether an "option" looks like a negative number
-        self._negative_number_matcher = _re.compile(r'^-\d[\d_]*(\.\d[\d_]*)?$')
+        self._negative_number_matcher = _re.compile(r'^-(\d+(_\d+)*(\.\d+(_\d+)*)?|\.\d+(_\d+)*)$')
```

Comment:
It is better to use non-capturing groups here: `(?:...)` instead of `(...)`. We do not need capturing groups, and non-capturing groups are faster.

And yes, the new regexp better matches the syntax of Python numbers. I was going to propose similar change in a review, but you already did this.

Reply:
Serhiy's note as a GH suggestion:
```suggestion
        self._negative_number_matcher = _re.compile(r'^-(?:\d+(?:_\d+)*(?:\.\d+(?:_\d+)*)?|\.\d+(?:_\d+)*)$')
```

Reply:
Thank you!